library/std/src/panicking.rs

```diff
@@ -725,20 +723,17 @@ fn rust_panic_with_hook(
             panic_count::MustAbort::AlwaysAbort => {
                 // Unfortunately, this does not print a backtrace, because creating
                 // a `Backtrace` will allocate, which we must to avoid here.
-                let panicinfo = PanicInfo::internal_constructor(
-                    message,
-                    location,
-                    can_unwind,
-                    force_no_backtrace,
-                );
-                rtprintpanic!("{panicinfo}\npanicked after panic::always_abort(), aborting.\n");
+                if let Some(message) = message {
+                    rtprintpanic!("aborting due to panic at {location}:\n{message}\n");
+                } else {
+                    rtprintpanic!("aborting due to panic at {location}\n");
+                }
             }
         }
         crate::sys::abort_internal();
     }
 
-    let mut info =
-        PanicInfo::internal_constructor(message, location, can_unwind, force_no_backtrace);
+    let mut info = PanicInfo::internal_constructor(location, can_unwind, force_no_backtrace);
```

Comment:
```suggestion
    // We will set a payload before passing this to the hook.
    let mut info = PanicInfo::internal_constructor(location, can_unwind, force_no_backtrace);
```

Reply:
I've removed the `set_payload` function and instead just set it all directly through the constructor, since there is no more need for a std PanicInfo to exist without a payload.